src/routers/course.ts

```diff
@@ -1,24 +1,65 @@
 import express from 'express';
+import { create, read, update, del } from '../operators/courseOperations';
 
 const courseRouter = express.Router();
+const app = express();
+
+app.use(express.json());
+
+courseRouter.get('/:id', async (req, res) => {
+	try {
+		const id = req.params.id;
+		const userID = req.header('userID');
+
+		if (id && userID) {
+			const course = await read(id, userID);
+			res.send(course);
```

Comment:
Does this work if the `courseItems` set is empty? 
I struggled with this in the `semesterRouter.get` function and the work around was to convert the courses set to an array before sending:
```
const resData = {
	...semester,
	courses: Array.from(semester?.courses || []),
};
res.send(resData);
```

Reply:
This could be the issue we were having with the read function not working sometimes too. Let's follow what Amanda did.